src/displayapp/screens/WatchFacePrideFlag.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include <lvgl/src/lv_core/lv_obj.h>
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <displayapp/Controllers.h>
+#include "displayapp/screens/Screen.h"
+#include "displayapp/widgets/StatusIcons.h"
+#include "components/datetime/DateTimeController.h"
+#include "components/ble/BleController.h"
+#include "utility/DirtyValue.h"
+
+namespace Pinetime {
+  namespace Controllers {
+    class Settings;
+    class Battery;
+    class Ble;
+    class NotificationManager;
+    class MotionController;
+  }
+
+  namespace Applications {
+    namespace Screens {
+
+      class WatchFacePrideFlag : public Screen {
+      public:
+        WatchFacePrideFlag(Controllers::DateTime& dateTimeController,
+                           const Controllers::Battery& batteryController,
+                           const Controllers::Ble& bleController,
+                           Controllers::NotificationManager& notificationManager,
+                           Controllers::Settings& settingsController,
+                           Controllers::MotionController& motionController);
+        ~WatchFacePrideFlag() override;
+
+        bool OnTouchEvent(TouchEvents event) override;
+        bool OnButtonPushed() override;
+
+        void Refresh() override;
+
+        void UpdateSelected(lv_obj_t* object, lv_event_t event);
+
+        void UpdateScreen(Pinetime::Controllers::Settings::PrideFlag);
+
+      private:
+        Utility::DirtyValue<uint8_t> batteryPercentRemaining {};
+        Utility::DirtyValue<bool> powerPresent {};
+        Utility::DirtyValue<bool> bleState {};
+        Utility::DirtyValue<std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds>> currentDateTime {};
+        Utility::DirtyValue<uint32_t> stepCount {};
+        Utility::DirtyValue<bool> notificationState {};
+        Utility::DirtyValue<std::chrono::time_point<std::chrono::system_clock, std::chrono::days>> currentDate;
+        // Must be wrapped in a dirty value, since it is displayed in the day but is updated twice a day
+        Utility::DirtyValue<const char*> ampmChar {"AM"};
+
+        static Pinetime::Controllers::Settings::PrideFlag GetNext(Controllers::Settings::PrideFlag prideFlag);
+        static Pinetime::Controllers::Settings::PrideFlag GetPrevious(Controllers::Settings::PrideFlag prideFlag);
+
+        uint32_t savedTick = 0;
+        bool initialized = false;
+
+        lv_obj_t** backgroundSections;
```

Comment:
I don't like manual memory management. It's much too error prone to create a memory leak. please replace with a std::vector or similar